client/packages/system/src/Location/ListView/AppBarButtons.tsx

```diff
@@ -13,29 +12,30 @@ import {
   EnvUtils,
   Platform,
 } from '@openmsupply-client/common';
-import { useLocation } from '..';
+import { LocationRowFragment } from '..';
 import { locationsToCsv } from '../../utils';
 
 interface AppBarButtonsProps {
   onCreate: () => void;
+  locations?: LocationRowFragment[];
+  reportIsLoading: boolean;
 }
 
-export const AppBarButtons: FC<AppBarButtonsProps> = ({ onCreate }) => {
+export const AppBarButtons = ({
+  onCreate,
+  locations,
+  reportIsLoading,
+}: AppBarButtonsProps) => {
   const { success, error } = useNotification();
   const t = useTranslation();
-  const { isLoading, fetchAsync } = useLocation.document.listAll({
-    key: 'name',
-    direction: 'asc',
-  });
 
   const csvExport = async () => {
-    const data = await fetchAsync();
-    if (!data || !data?.nodes.length) {
+    if (!locations) {
       error(t('error.no-data'))();
       return;
     }
 
-    const csv = locationsToCsv(data.nodes, t);
+    const csv = locationsToCsv(locations, t);
```

Comment:
As per that issue. Items shown in UI should be converted to csv instead of everything